src/extensions/default/bramble/lib/PostMessageTransportRemote.js

```diff
@@ -38,11 +38,15 @@
         /**
          * Sends a message over the transport.
          * @param {string} msgStr The message to send.
+         * @param {string} data The data to be shown in console
+         * @param {string} level The console type used
          */
-        send: function (msgStr) {
+        send: function (msgStr, data, level) {
```

Comment:
This is an example of a "leaky abstraction."  You have code elsewhere that knows what a `level` is, and here you're exposing that lower level code's concepts.  As you move from the `console` (where `level` has meaning) to the more generic `transport`, where you are only sending data across a channel, you want to hide this detail.

In this case, you should merge the `level` into your `data`, such that you would call things like so:

```js
// before, leaking too much internal abstraction info
transport.send("bramble-console", args, type);

// after, with all data bundled into one object
transport.send("bramble-console", {args: args, type: type});
```

Now you can just send `data` across the wire, and let the `ConsoleManager` deal with the more complex object that has both `args` and `level` attached.  The rest of your code using the transport can be blissfully unaware.

Another side comment: in JS it would be `dataType` but never the pythonic `data_type`.